src/cudadev/CondFormats/SiPixelROCsStatusAndMapping.h

```diff
@@ -11,15 +13,15 @@ namespace pixelgpudetails {
   constexpr unsigned int MAX_SIZE_BYTE_BOOL = MAX_SIZE * sizeof(unsigned char);
 }  // namespace pixelgpudetails
 
-struct SiPixelROCsStatusAndMapping {
-  alignas(128) unsigned int fed[pixelgpudetails::MAX_SIZE];
-  alignas(128) unsigned int link[pixelgpudetails::MAX_SIZE];
-  alignas(128) unsigned int roc[pixelgpudetails::MAX_SIZE];
-  alignas(128) unsigned int rawId[pixelgpudetails::MAX_SIZE];
-  alignas(128) unsigned int rocInDet[pixelgpudetails::MAX_SIZE];
-  alignas(128) unsigned int moduleId[pixelgpudetails::MAX_SIZE];
-  alignas(128) unsigned char badRocs[pixelgpudetails::MAX_SIZE];
-  alignas(128) unsigned int size = 0;
-};
+generate_SoA_store(SiPixelROCsStatusAndMapping,
+  SoA_column(unsigned int, fed),
+  SoA_column(unsigned int, link),
+  SoA_column(unsigned int, roc),
+  SoA_column(unsigned int, rawId),
+  SoA_column(unsigned int, rocInDet),
+  SoA_column(unsigned int, moduleId),
+  SoA_column(unsigned char, badRocs),
+  SoA_scalar(unsigned int, size)
+);
```

Comment:
Even then the data could be read in from a binary file with the current struct definition (that could be even made internal to the `SiPixelROCsStatusAndMappingWrapperESProducer`), and then copied to a data structure that is more efficient for reads. But as long as the bit representation stays the same, I suppose it will continue to work.

Reply:
I reverted this change, and we now have a view directly defined with a column by column constructor (this feature was added during the hackathon).